include/magic_enum.hpp

```diff
@@ -222,14 +222,14 @@ class static_string<0> {
 
   constexpr explicit static_string(string_view) noexcept {}
 
-  constexpr const char* data() const noexcept { return nullptr; }
+  [[maybe_unused]] [[nodiscard]] constexpr const char* data() const noexcept { return nullptr; }
```

Comment:
I'm a little surprised by this construction, as I know it contradicts each other.

Reply:
`[[nodiscard]]` means that if you use it, the return value cannot be discarded.
`[[maybe_unused]]` means that there is a chance to this function is not used.
These notations are not contradicting each other.

If someone did not use this function, vs will warn about it, this is why `[[maybe_unused]]` needs because no one is used usually. 
If any code calls this function, that will generate a warning if it discards the return value. So:
```
int main() {
  static_string<0> ss{""};
  ss.data(); // the return value not used, this will generate a warning.
}
```

Reply:
And what warnings does it fix?

Reply:
I mean, it's all internal function, do we real need this? @schaumb

Reply:
I use clion as the development environment and the built-in inspection tools recognize that these const member functions have no external effect, so it only makes sense to call because of the return value. If the return value is not used, it makes no sense to call the function. The `[[nodiscard]]` tag was automatically added because of this recommendation.

This tag can be omitted because it is not strictly part of the issue.

Reply:
The `[[maybe_unused]]` tag fixes the following warnings:

```
https://raw.githubusercontent.com/Neargye/magic_enum/master/include/magic_enum.hpp(225): warning C4514: 'magic_enum::detail::static_string<0>::data': unreferenced inline function has been removed
https://raw.githubusercontent.com/Neargye/magic_enum/master/include/magic_enum.hpp(229): warning C4514: 'magic_enum::detail::static_string<0>::operator std::string_view': unreferenced inline function has been removed
https://raw.githubusercontent.com/Neargye/magic_enum/master/include/magic_enum.hpp(232): warning C4514: 'magic_enum::detail::pretty_name': unreferenced inline function has been removed
https://raw.githubusercontent.com/Neargye/magic_enum/master/include/magic_enum.hpp(259): warning C4514: 'magic_enum::detail::case_insensitive::to_lower': unreferenced inline function has been removed
https://raw.githubusercontent.com/Neargye/magic_enum/master/include/magic_enum.hpp(275): warning C4514: 'magic_enum::detail::find': unreferenced inline function has been removed

```

Reply:
WIll check

Reply:
The `[[maybe_unused]]` is standardized, and it's part of the C++ language. the `#pragma warning` is not. 
The warning has right, these functions are not used. I think this tag fits here, especially because these are internal stuff.

I can replace it in this PR, but I have feelings against it.

Reply:
But we use this functions)

Reply:
We don't use this function (based on the compiler).

These functions are `static_string<0>` members whose are only used if some enum has no name, or is not recognized.
`static_string<i>::data` is used only at `static_string<i>::operator string_view`, but `static_string<0>::operator string_view` returns a default constructed value, so nowhere used `data` and `size`.
 
The other 'free' functions are only used if the `magic_enum` specific functions are used, and template-specialized with some enum.